qiskit/pulse/schedule/pulse_schedule.py

```diff
@@ -91,34 +96,42 @@ def name(self) -> str:
     def channels(self) -> ChannelStore:
         return self._channel_store
 
-    def add(self,
-            commands: Union[PulseCommand, List[PulseCommand]],
-            channel: PulseChannel,
-            start_time: int):
-        """Add new pulse command(s) with channel and start time context.
+    def append(self, command: PulseCommand, channel: PulseChannel):
+        """Append a new pulse command on a channel at the timing
+        just after the last command finishes on the channel.
+
+        Args:
+            command (PulseCommand):
+            channel (PulseChannel):
+        """
+        try:
+            start_time = self.end_time_by(channel)  # TODO: need to add buffer?
+            self.add(TimedPulse(command, channel, start_time))
+        except ScheduleError as err:
+            logger.warning("Fail to append %s to %s", command, channel)
+            raise ScheduleError(err.message)
+
+    def insert(self, command: PulseCommand, channel: PulseChannel, start_time: int):
+        """Insert new pulse command with `channel` at `start_time`.
 
         Args:
-            commands (PulseCommand|list):
+            command (PulseCommand):
             channel:
             start_time:
         """
-        if isinstance(commands, PulseCommand):
-            try:
-                self.add_block(TimedPulse(commands, channel, start_time))
-            except ScheduleError as err:
-                logger.warning("Fail to add %s to %s at %s", commands, channel, start_time)
-                raise ScheduleError(err.message)
-        elif isinstance(commands, list):
-            for cmd in commands:
-                self.add(cmd, channel, start_time)
-
-    def add_block(self, block: TimedPulseBlock):
+        try:
+            self.add(TimedPulse(command, channel, start_time))
+        except ScheduleError as err:
+            logger.warning("Fail to insert %s to %s at %s", command, channel, start_time)
+            raise ScheduleError(err.message)
+
+    def add(self, block: TimedPulseBlock):
```

Comment:
Why do we need both `insert` and `add`? If we really do require it should be private, eg., `_add`.

Reply:
done